src/reducers/url-state.js

```diff
@@ -250,49 +250,75 @@ function pathInZipFile(
   }
 }
 
-const urlStateReducer: Reducer<UrlState> = (regularUrlStateReducer => (
-  state: UrlState,
-  action: Action
-): UrlState => {
-  switch (action.type) {
-    case 'UPDATE_URL_STATE':
-      return action.newUrlState;
-    default:
-      return regularUrlStateReducer(state, action);
-  }
-})(
+/**
+ * These values are specific to an individual profile.
+ */
+const profileSpecific = combineReducers({
+  rangeFilters,
+  selectedThread,
+  callTreeSearchString,
+  threadOrder,
+  hiddenThreads,
+  markersSearchString,
+  transforms,
+});
+
+/**
+ * Provide a mechanism to wrap the UrlState reducer in a special function that can swap
+ * out the entire UrlState with a new one coming from the History API. Also provide a
+ * way to invalidate sections of the state based off of looking at multiple profiles.
+ */
+const wrapReducerInResetter = (
+  regularUrlStateReducer: Reducer<UrlState>
+): Reducer<UrlState> => {
+  return (state, action) => {
+    switch (action.type) {
+      case 'UPDATE_URL_STATE':
+        // A new URL came in because of a browser action, discard the current UrlState
+        // and use the new one, which was probably serialized from the URL, or stored
+        // in the history API.
+        return action.newUrlState;
+      case 'RETURN_TO_ZIP_FILE_LIST':
+        // Invalidate all information that would be specific to an individual profile.
+        return Object.assign(regularUrlStateReducer(state, action), {
+          profileSpecific: profileSpecific(undefined, state),
+        });
+      default:
+        return regularUrlStateReducer(state, action);
+    }
+  };
+};
```

Comment:
So here I'd keep the previous wrapper for `UPDATE_URL_STATE` only as it was, and use the aforementioned extracted `wrapReducerInResetter` for the inner reducer `profileSpecific` only:
```javascript
const profileSpecific = wrapReducerInResetter(
  'RETURN_TO_ZIP_FILE_LIST',
  combineReducers({
    rangeFilters,
    selectedThread,
    callTreeSearchString,
    threadOrder,
    hiddenThreads,
    markersSearchString,
    transforms,
  })
);
```